### ISSUE TITLE
Website: Multi-Calendar & Pricing main (CRU)

### ISSUE THREAD

**Issue by stefanhopman1** [domits1/Domits#36]
## 🚀 Feature Summary

The Calendar & Pricing feature provides centralized control over availability, rates, and booking rules. It enables real-time updates, dynamic pricing, and rule-based adjustments to maximize occupancy and revenue. Users can manage seasons, minimum stays, and promotions with full visibility across all properties and channels.

There is an availability calendar for the host and a booking calendar for the guest.

----

Epics: [PMS](https://github.com/domits1/Domits/issues/149), [CRS](https://github.com/domits1/Domits/issues/501), [CM](https://github.com/domits1/Domits/issues/440), [RMS](https://github.com/domits1/Domits/issues/436), [Calendar & Prices](https://github.com/domits1/Domits/issues/36), [Inbox](https://github.com/domits1/Domits/issues/767), [Finance](https://github.com/domits1/Domits/issues/163), [Integrations](https://github.com/domits1/Domits/issues/433)
Use Case / Scenario Testing: [PMS](https://github.com/domits1/Domits/issues/2262), [CRS](https://github.com/domits1/Domits/issues/2265), [CM](https://github.com/domits1/Domits/issues/2266), [RMS](https://github.com/domits1/Domits/issues/2267), [Calendar & Prices](https://github.com/domits1/Domits/issues/2299), [Inbox](https://github.com/domits1/Domits/issues/2291), [Finance](https://github.com/domits1/Domits/issues/2287)

----

<img width="1306" height="517" alt="Image" src="https://github.com/user-attachments/assets/18831730-9e64-428b-8897-7acc438ba687" />

----

# Main issue

**Calendar & Pricing Host DASHBOARD:**
- [ ] #766 

**Calendar & Pricing GUEST experience:**
- [ ] #1331

**Calendar & Pricing Host ONBOARDING:**
- [x] #1220 

---

## 🧩 Problem Statement
What problem does this feature solve? Who is affected, and why does it matter? > Focus on the problem, not the solution.

No centralized multi-calendar and pricing

---

## 🎯 Desired Outcome
What does success look like if this feature is implemented?

Centralized multi-calendar and pricing

---

## 👥 Target Users
Who benefits from this feature?
- [ ] Guests / Customers
- [x] Hosts / Property Managers / Enterprise
- [ ] Admin / Internal users
- [ ] Developers
- [ ] Other: _______

---

## 📈 Business & User Impact
How valuable is this feature?

- **Impact level:**
  - [ ] Low (nice-to-have)
  - [ ] Medium (improves workflows)
  - [ ] High (unblocks key use cases)
  - [x] Critical (strategic / revenue-impacting)

- **User reach:**
  - [ ] Few users
  - [ ] Many users
  - [x] All users
  - [ ] Unknown

---

## ⚠️ Constraints & Considerations
- Performance
- Security
- Privacy
- Accessibility
- Backward compatibility
- External dependencies

---

## 🧪 Acceptance Criteria / User Stories
How can we validate that this feature is complete?

- [ ] Given X, when Y, then Z

👥 User Roles

- Host / Property Manager
- Revenue Manager / Pricing Analyst
- Reservation Agent
- Admin / System Operator
- Channel Manager / OTA Integration System

📅 1. Calendar Management – Availability & Bookings

- [x] As a host, I want to view a unified calendar for all my properties so I can see bookings and availability at a glance.
- [x] As a host, I want to block specific dates for maintenance or personal use.
- [ ] As a system, I want to automatically mark dates as unavailable after a booking is confirmed.
- [ ] As a host, I want to set check-in and check-out days restrictions per property.
- [ ] As a host, I want to prevent overbooking by syncing availability across multiple channels.
- [ ] As a host, I want to view calendar conflicts and resolve them before confirming new bookings.
- [ ] As a host, I want to set advance notice rules (minimum lead time before booking).
- [ ] As a host, I want to set maximum stay limits for specific seasons.
- [ ] As a host, I want to see a daily, weekly, and monthly calendar view.
- [ ] As a host, I want color-coded calendar blocks to differentiate bookings, blocked dates, and maintenance.
- [ ] As a host, I want recurring block dates (e.g., every Monday unavailable) for routine maintenance.
- [ ] As a system, I want to automatically update calendar availability after a cancellation.
- [ ] As a revenue manager, I want to view occupancy trends over time.
- [ ] As a host, I want to assign bookings to specific units within a property.
- [ ] As a host, I want to merge calendar data from multiple sources (PMS, OTA, direct bookings).

🔄 2. Multi-Channel Calendar Sync

- [ ] As a host, I want to connect multiple OTA accounts (Airbnb, Booking.com, VRBO, Expedia) to sync availability.
- [ ] As a system, I want to automatically push availability updates to all connected OTAs.
- [ ] As a host, I want to pull existing bookings from OTAs into my central calendar.
- [ ] As a host, I want to pause or unpublish a property on specific channels temporarily.
- [ ] As a host, I want to see which channels a property is connected to and their sync status.
- [ ] As a system, I want to handle sync failures and retries automatically.
- [ ] As an admin, I want logs of all calendar syncs with timestamps and status.
- [ ] As a host, I want to prevent double bookings by locking dates immediately when a reservation is made.
- [ ] As a host, I want to reconcile calendar discrepancies between PMS and OTAs.
- [ ] As a host, I want to apply channel-specific booking rules (minimum stay, lead time) per OTA.

💵 3. Base Rates & Seasonal Pricing

- [x] As a host, I want to set base rates for each property/unit.
- [ ] As a revenue manager, I want to define seasonal pricing for high and low-demand periods.
- [ ] As a system, I want to apply rate changes automatically for specified dates.
- [ ] As a host, I want to override base rates for specific dates.
- [ ] As a revenue manager, I want to create special event pricing (holidays, festivals, conferences).
- [ ] As a host, I want to set daily, weekly, and monthly rate rules.
- [ ] As a system, I want to handle different currencies per property or channel.
- [ ] As a host, I want to see rate history to track pricing changes.
- [ ] As a revenue manager, I want to compare current rates against competitors or market benchmarks.
- [ ] As a host, I want to schedule future price changes in advance.

⚖️ 4. Dynamic & Advanced Pricing Rules

- [ ] As a revenue manager, I want dynamic pricing based on occupancy (higher rates when occupancy is high).
- [ ] As a host, I want minimum stay rules applied automatically based on season or booking dates.
- [ ] As a host, I want maximum stay rules applied automatically to avoid long-term unprofitable bookings.
- [ ] As a system, I want to automatically increase rates for last-minute bookings.
- [ ] As a system, I want to apply length-of-stay discounts automatically (e.g., 10% off for stays 7+ nights).
- [ ] As a revenue manager, I want to adjust rates by channel to account for commissions.
- [ ] As a host, I want to apply blackout dates where certain discounts or rates do not apply.
- [ ] As a system, I want to prevent conflicting rate rules from overlapping.
- [ ] As a revenue manager, I want to simulate different pricing scenarios to predict revenue.
- [ ] As a system, I want to integrate with market demand APIs for suggested rates.

🧾 5. Promotions & Discounts

- [ ] As a host, I want to create promotional rates for off-season periods.
- [ ] As a system, I want to automatically apply early-bird discounts.
- [ ] As a system, I want to automatically apply last-minute booking discounts.
- [ ] As a host, I want to apply promo codes for direct bookings.
- [ ] As a host, I want to limit promotions to specific channels.
- [ ] As a revenue manager, I want to set minimum stay requirements for discounts.
- [ ] As a host, I want to track the effectiveness of promotions in increasing occupancy.
- [ ] As a system, I want to prevent overlapping promotions that conflict with base rates.
- [ ] As a host, I want to set group booking discounts automatically.
- [ ] As a host, I want to apply special pricing for long-term stays (1 month+).

🔔 6. Calendar Alerts & Notifications

- [ ] As a host, I want to receive alerts when bookings overlap or conflict with blocked dates.
- [ ] As a revenue manager, I want notifications when occupancy drops below a threshold.
- [ ] As a host, I want alerts for un-synced channels or failed updates.
- [ ] As a system, I want to notify users of rate changes or promotions applied automatically.
- [ ] As a host, I want reminders for upcoming seasonal price changes.
- [ ] As a system, I want to trigger alerts for manually overridden rates.
- [ ] As a host, I want automated warnings for double-booking attempts.

📊 7. Reporting & Analytics

- [ ] As a system, I want to export calendar and pricing reports to Excel or CSV.

🔄 8. Multi-Property & Bulk Management

- [ ] As a host, I want to update rates for multiple properties simultaneously.
- [ ] As a host, I want to apply seasonal pricing rules across all units in a property portfolio.
- [ ] As a system, I want to replicate calendar and pricing changes across multiple channels automatically.
- [ ] As a revenue manager, I want to clone rate plans from one property to another.
- [ ] As a host, I want bulk block/unblock dates for multiple properties.
- [ ] As a system, I want to prevent bulk actions that would create booking conflicts.
- [ ] As a host, I want to apply promotions to multiple units or properties at once.

🔐 9. Security & Permissions

- [ ] As an admin, I want role-based access for calendar and pricing changes.
- [ ] As a system, I want to maintain logs of all calendar and rate modifications.
- [ ] As a host, I want to approve rate changes before they are pushed live.
- [ ] As an admin, I want to restrict access to dynamic pricing tools to authorized users.
- [ ] As a system, I want audit trails for promotions and seasonal pricing.

🧩 10. Integration & APIs

- [ ] As a system, I want to integrate with OTAs (Airbnb, Booking.com, VRBO) for two-way calendar sync.
- [ ] As a system, I want APIs to allow third-party revenue management software to update rates.
- [ ] As a host, I want iCal support for external calendar integrations.
- [ ] As a system, I want to detect and handle API failures or delayed updates.
- [ ] As an admin, I want logs for all API-based calendar and pricing updates.

**Comment by stefanhopman1** [comment]
https://github.com/domits1/Domits/issues/154
